source/tunneling/SecureTunnelingContext.cpp

```diff
@@ -33,7 +33,16 @@ namespace Aws
                 {
                 }
 
-                SecureTunnelingContext::~SecureTunnelingContext() { mSecureTunnel->Close(); }
+                SecureTunnelingContext::~SecureTunnelingContext() {
+                    LOG_DEBUG(TAG, "SecureTunnelingContext::~SecureTunnelingContext");
+//                    mSecureTunnel->Close();
```

Comment:
Hey @marcoemorais-aws , If in call `Close()` over here, I am seeing a segmentation fault. Any thoughts on that?

Reply:
I am able to reproduce this locally and looks like we are passing a null pointer to aws_secure_tunnel_close — the Aws::Iotsecuretunneling::SecureTunnel::Close should have a null check.  Please add the snippet below and the issue should be resolved (alternately you can use `mSecureTunnel->IsValid()` as a substitute for `mSecureTunnel->GetUnderlyingHandle()`).

```
                    if (mSecureTunnel->GetUnderlyingHandle())
                    {
                        mSecureTunnel->Close();
                    }
```